src/main/java/commands/AddCommand.java

```diff
@@ -22,10 +24,14 @@ public AddCommand(String question, String answer) {
     }
 
     @Override
-    public void execute(CardList cards, Ui ui) {
-        cards.addCard(card);
-        int cardCount = cards.getCardCount();
-        ui.showCardAdded(cards.getCard(cardCount - 1), cardCount);
+    public void execute(CardList cards, Ui ui, Access access, Storage storage) {
+        if (access.getChapterLevel() != "") {
+            cards.addCard(card);
+            int cardCount = cards.getCardCount();
+            ui.showCardAdded(cards.getCard(cardCount - 1), cardCount);
+        } else {
+            System.out.println("Sorry, you currently are in the wrong level, please enter chapter level first.");
+        }
```

Comment:
instead of checking if (access.getChapterLevel() != "") first, you can do something like this:
if (access.getChapterLevel().equals("")) {
// print error message
return;
}
// rest of the code that should be executed